### DISCUSSION TITLE
Conceptual question: Does it make any sense to add a filter without a backing form field?

### DISCUSSION THREAD

**Discussion by is-pdoherty** [carltongibson/django-filter#1583] [Q&A]
I'd mentioned this in #1418 but I figured this question is more broadly applicable and I didn't want to hijack that thread.

What I'm wondering is: 

1. Does it make sense to add a filter for a model field which doesn't have a corresponding Django form?
2. If not, does it make sense to create a form for just this purpose (i.e. creating a custom Django filter subclass) or am I naively proceeding down the wrong path here? Should this be done via another means (e.g. queryset) instead?

I'm looking to create a custom filter for [Django PG Vector columns](https://github.com/pgvector/pgvector-python). My use case is somewhat complicated because it involves PG Vector and Graphene but, distilled, I think it would be being able to translate:

`Item.objects.order_by(L2Distance('embedding', [3, 1, 2]))[:5]`

into something like:

`ItemFilter(lookup_expr="order_by_l2_distance: [3, 1, 2]")`

in order to use it in graphql like:

```
query {
   allItems(order_by_l2_distance: [3, 1, 2]) {
    edges {
      node {
        id
      }
    }
  }
}

```

**Comment by carltongibson** [comment]
The form field (plus its widget)) maps from the query string to the values passed to the filter. 

So in your case you want multiple values passed: `[3, 1, 2]`

The query string for that would either be`?item=3&item=1&item=2` using a multiple choice filter (say, for discussion) or else `?item=3,1,2` using [a CSV filter](https://django-filter.readthedocs.io/en/stable/ref/filters.html#base-in-filter). 

The form field's job, really, is to exclude bad data, such as passing totally the wrong type: 

`?item=cat&item=hat&item=mat`

So even in a GraphQL case, you're still going to want to use one. (I'd never pass raw input from the request straight to my backend. The ORM &co are good at avoiding injection but basic sanitation is still required.)
